sim/warrior/shield_block.go

```diff
@@ -10,9 +10,24 @@ import (
 func (warrior *Warrior) RegisterShieldBlockCD() {
 	actionID := core.ActionID{SpellID: 2565}
 
+	T12_4P_bonus := warrior.RegisterAura(core.Aura{
+		Label:    "T12 4P Bonus",
+		ActionID: core.ActionID{SpellID: 99242},
+		Duration: 10 * time.Second,
+		OnGain: func(aura *core.Aura, sim *core.Simulation) {
+			warrior.AddStatDynamic(sim, stats.Parry, 6*core.ParryRatingPerParryChance)
+		},
+		OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+			warrior.AddStatDynamic(sim, stats.Parry, -6*core.ParryRatingPerParryChance)
+		},
+	})
```

Comment:
You should put this inside an if hasT12_4P_bonus so it doesnt register the aura without the set equipped. If you do this you dont need to check if hasT12 is true in the proc trigger because our aura.Activate() already does a null check for the aura you try to activate.

Reply:
Okay, so I added aura to Warrior.  If I understand you correctly I can involve it in Sheildblock expires just as a warrior.T12_4P_Bonus.activate(sim)  with the sim. duration checks correct?

Reply:
You dont need a new reference in warrior to this aura. Keep it a local variable in the register function and only assign it if the set is equipped. Then in the expire handler you only need to check if its end of iteration and if not activate the aura. Even if the set is not equipped it will still work with a null aura, it just wont try to activate

Reply:
Okay, I think I did what you wanted.  Let me know if I missed the mark.